.github/workflows/main.yml

```diff
@@ -1,21 +1,115 @@
-name: Run tree-sitter tests
+name: CI
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
+    paths:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
   pull_request:
-    branches: [ master ]
+    branches: [master]
+    paths:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
+
+concurrency:
+  group: ${{github.workflow}}-${{github.ref}}
+  cancel-in-progress: true
 
 jobs:
   test:
     name: Tree-sitter tests
-    runs-on: ubuntu-latest
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-14]
     steps:
-      - uses: actions/checkout@v4
-
-      - uses: actions/setup-node@v4
-      - name: Install Dependencies
-        run: npm install
-
+      - name: Set up repository
+        uses: tree-sitter/parser-setup-action@v1.1
+        with:
+          node-version: 20
+      - name: Clone nvim help files
+        uses: actions/checkout@v4
+        with:
+          repository: neovim/neovim
+          path: examples/neovim
+          sparse-checkout: runtime/doc/
       - name: Run tests
-        run: npm test
+        uses: tree-sitter/parser-test-action@v1.2
+        with:
+          test-library: ${{runner.os == 'Linux'}}
+          corpus-files: |-
+            examples/neovim/runtime/doc/*
+          # FIXME: these files should not have errors
+          invalid-files: |-
```

Comment:
Presumably the "errors" are the ones ignored by `gen_help_html.lua`: https://github.com/neovim/neovim/blob/bbb68e2a034ad3aaea99178c09301ca458ee8dff/scripts/gen_help_html.lua#L329-L331

Likely the best solution is to fix the help files themselves.